File: apps/server-go/internal/model/access.go
package model

import "time"

// Permission 对应 permissions 表，表示一个可被角色授予的原子权限。
type Permission struct {
	ID          int64     `db:"id"`
	Code        string    `db:"code"`
	Module      string    `db:"module"`
	Action      string    `db:"action"`
	Name        string    `db:"name"`
	Description *string   `db:"description"`
	CreatedAt   time.Time `db:"created_at"`
}

// Role 对应 roles 表，表示一组权限集合。
type Role struct {
	ID          int64     `db:"id"`
	Code        string    `db:"code"`
	Name        string    `db:"name"`
	Description *string   `db:"description"`
	IsSystem    bool      `db:"is_system"`
	SortOrder   int       `db:"sort_order"`
	CreatedAt   time.Time `db:"created_at"`
	UpdatedAt   time.Time `db:"updated_at"`
}

// Team 对应 teams 表，表示协作团队。
type Team struct {
	ID          int64     `db:"id"`
	Name        string    `db:"name"`
	Slug        string    `db:"slug"`
	Description *string   `db:"description"`
	OwnerID     *int64    `db:"owner_id"`
	Visibility  string    `db:"visibility"`
	CreatedBy   *int64    `db:"created_by"`
	CreatedAt   time.Time `db:"created_at"`
	UpdatedAt   time.Time `db:"updated_at"`
}

// TeamMember 对应 team_members 表，表示团队成员关系。
type TeamMember struct {
	TeamID     int64     `db:"team_id"`
	UserID     int64     `db:"user_id"`
	MemberRole string    `db:"member_role"`
	Status     string    `db:"status"`
	AddedBy    *int64    `db:"added_by"`
	JoinedAt   time.Time `db:"joined_at"`
}

// ContentShare 对应 content_shares 表，表示对文章/媒体资源的共享授权。
type ContentShare struct {
	ID              int64      `db:"id"`
	ResourceType    string     `db:"resource_type"`
	ResourceID      int64      `db:"resource_id"`
	PrincipalType   string     `db:"principal_type"`
	PrincipalID     int64      `db:"principal_id"`
	PermissionLevel string     `db:"permission_level"`
	GrantedBy       *int64     `db:"granted_by"`
	ExpiresAt       *time.Time `db:"expires_at"`
	CreatedAt       time.Time  `db:"created_at"`
	UpdatedAt       time.Time  `db:"updated_at"`
}

File: packages/types/src/models/access.ts
export type SystemRoleCode = 'ADMIN' | 'AUTHOR' | 'USER';

export type UserStatus = 'ACTIVE' | 'INACTIVE' | 'BANNED';

export interface Permission {
  id: number;
  code: string;
  module: string;
  action: string;
  name: string;
  description?: string;
}

export interface Role {
  id: number;
  code: SystemRoleCode | string;
  name: string;
  description?: string;
  isSystem: boolean;
  sortOrder: number;
  permissions: Permission[];
}

export interface ManagedUser {
  id: number;
  username: string;
  email: string;
  nickname?: string;
  avatar?: string;
  bio?: string;
  role: SystemRoleCode | string;
  roles: string[];
  status: UserStatus;
  mustChangePassword: boolean;
  lastLoginAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateManagedUserRequest {
  username: string;
  email: string;
  password: string;
  nickname?: string;
  roleCodes?: string[];
  status?: UserStatus;
  mustChangePassword?: boolean;
}

export interface UpdateManagedUserRequest {
  email?: string;
  nickname?: string;
  bio?: string;
  status?: UserStatus;
  roleCodes?: string[];
  mustChangePassword?: boolean;
}

export interface Team {
  id: number;
  name: string;
  slug: string;
  description?: string;
  ownerId?: number;
  visibility: 'PRIVATE' | 'INTERNAL' | 'PUBLIC';
  memberCount: number;
  createdBy?: number;
  createdAt: string;
  updatedAt: string;
}

export interface TeamMember {
  teamId: number;
  userId: number;
  username: string;
  nickname?: string;
  email: string;
  memberRole: 'OWNER' | 'MANAGER' | 'MEMBER' | 'VIEWER';
  status: 'ACTIVE' | 'INVITED' | 'DISABLED';
  addedBy?: number;
  joinedAt: string;
}

export interface CreateTeamRequest {
  name: string;
  slug: string;
  description?: string;
  ownerId?: number;
  visibility?: Team['visibility'];
}

export type SharedResourceType = 'POST' | 'MEDIA_FILE' | 'MEDIA_FOLDER';
export type SharePrincipalType = 'USER' | 'TEAM' | 'ROLE';
export type ContentPermissionLevel = 'VIEW' | 'COMMENT' | 'EDIT' | 'MANAGE';

export interface ContentShare {
  id: number;
  resourceType: SharedResourceType;
  resourceId: number;
  principalType: SharePrincipalType;
  principalId: number;
  permissionLevel: ContentPermissionLevel;
  grantedBy?: number;
  expiresAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface ShareableResource {
  resourceType: SharedResourceType;
  id: number;
  title: string;
  subtitle?: string;
  status?: string;
  updatedAt?: string;
}

export interface CreateContentShareRequest {
  resourceType: SharedResourceType;
  resourceId: number;
  principalType: SharePrincipalType;
  principalId: number;
  permissionLevel: ContentPermissionLevel;
  expiresAt?: string;
}

export interface BatchCreateContentSharesRequest {
  resourceType: SharedResourceType;
  resourceIds?: number[];
  resourceSearch?: string;
  selectAllMatching?: boolean;
  principalType: SharePrincipalType;
  principalId: number;
  permissionLevel: ContentPermissionLevel;
  expiresAt?: string;
}

export interface BatchCreateContentSharesResult {
  total: number;
  shares: ContentShare[];
}
